string opt
uint32 id
string name
string type
uint32 floor
